Dockerfile

```diff
@@ -1,20 +1,29 @@
 FROM python:3.6
 
-# NodeJS's version is not pinned becuase nodesource only serve the latest
-# version.
-ENV YARN_VERSION 1.15.2-1
 ENV PYTHONUNBUFFERED 1
 ENV BASE_DIR /usr/local
 ENV APP_DIR $BASE_DIR/app
 
+ENV NVM_INSTALLER_URL https://raw.githubusercontent.com/creationix/nvm/v0.33.0/install.sh
+ENV NVM_DIR $BASE_DIR/nvm
+ENV YARN_VERSION 1.15.2-1
+ENV NODE_VERSION 8.16.0
+
 # Install Node and Yarn from upstream
-RUN curl -sS https://deb.nodesource.com/gpgkey/nodesource.gpg.key | apt-key add - \
- && echo 'deb http://deb.nodesource.com/node_8.x stretch main' | tee /etc/apt/sources.list.d/nodesource.list \
- && curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - \
- && echo 'deb http://dl.yarnpkg.com/debian/ stable main' | tee /etc/apt/sources.list.d/yarn.list \
- && apt-get update \
- && apt-get install -y nodejs yarn=$YARN_VERSION \
- && rm -rf /var/lib/apt/lists/*
+RUN curl -o- $NVM_INSTALLER_URL | bash \
+ && . $NVM_DIR/nvm.sh \
+ && nvm install $NODE_VERSION \
+ && nvm alias default $NODE_VERSION \
+ && nvm use default \
+ && nvm --version
+RUN . $NVM_DIR/nvm.sh \
+ && npm install -g yarn \
+ && yarn --version \
+# make nodejs and yarn accessible and executable globally
+ENV NODE_PATH $NVM_DIR/v$NODE_VERSION/lib/node_modules
+ENV PATH $NVM_DIR/versions/node/v$NODE_VERSION/bin:$PATH
```

Comment:
I would prefer setting `PATH` earlier and avoiding `. nvm.sh` repeatedly in each `RUN`.

Reply:
1. Repeating `. nvm.sh` is for debugging by using two `RUN` rather than a big chunk pipeline of commands. It is optional if the recipe is stable enough. So +1 to remove the repeated `. nvm.sh`.

2. +1 to make `PATH` earlier. I made the other `PATH` setup earlier as well while fixing the code.